### PR TITLE
Some config enhancements for test suite and debugging/logging

### PR DESCRIPTION
**Pull request by hangtwenty**
I'm opening this **for discussion,** of course... I expect to have to touch things up. I also recognize this is more than usually goes in one Pull Request. Just wanted to have a discussion in one place. I can split it up into smaller changes with `cherry-pick` as desired 😉

🙇 Also!!! Kudos for a very clean codebase! And great testing+CI setup already with the test suite, tox, Travis, and more.

Rundown:

* Add test suite config to use `pytest-xdist` to parallelize across up to 4 cores, speeding up test suite by a bit better than 2x. [More notes in commit logs.](https://github.com/hangtwenty/eemeter/commit/6d574da726c3f383d028b8bb51abbc518afc3a7d)

* [Add versatile tools for controlling logging from the CLI](https://github.com/openeemeter/eemeter/pull/127/commits/754c93de5e7fb8ea3159e9a38bb5ef90ec78e646). Allows changing the whole logging config, but more typically you would let the default config get used... and optionally `--log-console` to turn on console log output, and/or `--log-level=DEBUG` to increase the verbosity. For the console logger and the `DEBUG` level logger, the default logging config's log format has a "debug trace" flavor to it, showing modules, function names, and line numbers. (Reusing my favorite base config from other projects.) 
    * [Here are some example console logs with `--log-level=DEBUG --log-console`](https://gist.github.com/hangtwenty/5145ac8c7a398f39dc0e3226c3c33d6e)

* During test suite, set to max verbosity and logs on console. This just results in more info when you want it. You won't see extra output except when tests fail.

* During test suite, disable file logging.

* Consolidate `pytest.ini` and `tox.ini`, [rationale in commit log](https://github.com/openeemeter/eemeter/pull/127/commits/429a05b11dda0475d9755849f5c58fcd9d9c2c23)

* set flake8 config to line lengths of 100, [rationale in this commit message](https://github.com/openeemeter/eemeter/pull/127/commits/03b861a0e228faa7a5ced67a5931fc40d2a4ab42)

* **No more Internet needed for the test suite!** Mocked NOAA FTP Responses (and set up a utility to update the fixture data). Details on this below :)

### CODE REVIEW

**Comment by hangtwenty** [comment]
p.s. I love [CodeClimate](http://codeclimate.com) for Python, and [I did set it up on my fork.](https://codeclimate.com/github/hangtwenty/eemeter/trends/churn). But it doesn't really make sense to set up from a fork, so I removed those commits. If you're curious what it could offer, here's an example of its findings:

* https://codeclimate.com/github/hangtwenty/eemeter/trends/churn

**Comment by hangtwenty** [comment]
## 👨‍🔬 fixtures for `test_noaa_client.py` (2748aa3)

Added a script to "capture" responses from NOAA and
help save those as a test suite fixture.

Then load in and mock one method in NOAAClient to load
from that previously saved file.

In spirit it's similar to the VCR.py library,
but that only works for HTTP/S, not FTP.

It might seem like a lot just to get these 2 test cases 'unplugged' ...  but now I think this opens up to add a variety of further test cases if desired.

Bottom line: test_noaa_client.py doesn't talk to the Internet anymore!

### Tests: Disable the Internet during tests. (3f44f3a)

* Depends on previous commit to make test_noaa_client.py
no longer need the Internet.

* Use `pytest-socket` to disable `socket.socket`,
which effectively disables all network access.
(Humblebrag: Incidentally `pytest-socket` is based on a
StackOverflow answer I gave a while back hehe.)

**Comment by philngo** [comment]
These are all great suggestions. I'm going to close this for now but create issues for each of these separately so that we can pull them in one at at time.
